test/classes/initializers/generics/overloading.chpl

```diff
@@ -14,11 +14,11 @@ class D : C {
     this.complete();
 
     for i in 1..rank do
-      ranges(i) = 1..i;
+      ranges(i-1) = 1..i;
   }
 
   override proc bbox(x: int) {
-    return ranges(x);
+    return ranges(x-1);
```

Comment:
I viewed this test as testing overloading/overriding rather than computation of dimensions and bounding boxes, so took the minimal change required.  I'll update it, but it seems unlikely that people will be spending a lot of time browsing and reading this test in the years to come, nor that the current behavior is all that confusing.

Reply:
Here are the changes I made:

```diff
index 756aca1482..6851949e93 100644
--- a/test/classes/initializers/generics/overloading.chpl
+++ b/test/classes/initializers/generics/overloading.chpl
@@ -13,18 +13,18 @@ class D : C {
 
     this.complete();
 
-    for i in 1..rank do
-      ranges(i-1) = 1..i;
+    for i in 0..<rank do
+      ranges(i) = 1..i+1;
   }
 
   override proc bbox(x: int) {
-    return ranges(x-1);
+    return ranges(x);
   }
 }
 
 var d: C = new D(4);
 
+writeln(d.bbox(0));
 writeln(d.bbox(1));
 writeln(d.bbox(2));
 writeln(d.bbox(3));
-writeln(d.bbox(4));
```

Reply:
Awesome, thanks!